Sources/BrowserServicesKit/ContentBlocking/ContentBlockerRulesManager.swift

```diff
@@ -388,6 +386,15 @@ public class ContentBlockerRulesManager: CompiledRuleListsSource {
                                                                                                 unprotectedSitesHash: nil))
             }
 
+            if let compilationTime = result.compilationTime {
+                
+                // todo: map broken sources to iteration count
+                let iteration = task.sourceManager.brokenSources
```

Comment:
I'd consider keeping that logic in Task itself - to encapsulate usage of `brokenSources` there (I know it is currently not fully encapsulated, but I'd still limit the 'external' usage for better readability).